### PR TITLE
adding key separator into spec to allow '.' in key values

### PR DESCRIPTION
**Pull request by henry-megarry**

### CODE REVIEW

**Comment by JoshKCarroll** [comment]
Hey @henry-megarry, thanks for this PR! I can dig this change. I was wondering if you would be willing to make two changes:
* Add some additional test cases, both testing the utils for non-`.` keySeparator values as well as a couple of integration tests in `kazaam_int_test.go` with both explicitly setting `.` and setting non-standard keySeparator values.
* I'm wondering if we could do the setting of default keySeparator during initialization instead of hard coding it into all the util functions. Perhaps inspecting and updating the config in the [custom UnmarshalJSON() that we already have](https://github.com/qntfy/kazaam/blob/v3.4.7/spec.go#L21-L37)?

Appreciate it!!

**Comment by henry-megarry** [comment]
cool I have added a few more tests.

I have also changed it to default it in the unmarshalJSON method. some of the tests use the getConfig() method in util_tests.go that I had to update to add "." as the key separator, just want to make sure thats alright?

File: transform/util_test.go [hunk 1]
```diff
@@ -104,7 +104,35 @@ func TestSetJSONRaw(t *testing.T) {
...
+		areEqual, _ := checkJSONBytesEqual(actual, testItem.expectedOutput)
+		if !areEqual {
+			t.Error("Error data does not match expectation.")
+			t.Log("Expected:   ", testItem.expectedOutput)
+			t.Log("Actual:     ", string(actual))
+		}
+	}
+}
+
+func TestSetJSONRawNonDefaultKeySeporator(t *testing.T) {
```
Comment (JoshKCarroll) [inline transform/util_test.go]: Minor typo `Seporator` -> `Separator`

**Comment by JoshKCarroll** [comment]
Yeah this looks great if you wanna just fix that typo ^ will approve and merge. Thanks!!

File: transform/util_test.go (hunk 1 above)
Comment (henry-megarry) [reply transform/util_test.go]: fixed

**Comment by JoshKCarroll** [comment]
Tagged `v3.4.8`

**Pull request merged**